test/new-e2e/pkg/components/remotehost.go

```diff
@@ -6,131 +6,28 @@
 package components
 
 import (
-	"context"
-	"fmt"
-	"io/fs"
-	"net"
-	"os"
-	"strings"
-	"time"
-
 	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
-	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/runner"
-	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/runner/parameters"
-	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/clients"
-	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/optional"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/client"
 
 	osComp "github.com/DataDog/test-infra-definitions/components/os"
 	"github.com/DataDog/test-infra-definitions/components/remote"
-
-	"github.com/stretchr/testify/require"
-	"golang.org/x/crypto/ssh"
-)
-
-const (
-	// Waiting for only 10s as we expect remote to be ready when provisioning
-	sshRetryInterval = 2 * time.Second
-	sshMaxRetries    = 20
 )
 
 // RemoteHost represents a remote host
 type RemoteHost struct {
 	remote.HostOutput
 
-	client  *ssh.Client
+	*client.Host
 	context e2e.Context
 }
 
-var _ e2e.Initializable = &RemoteHost{}
+var _ e2e.Initializable = (*RemoteHost)(nil)
 
 // Init is called by e2e test Suite after the component is provisioned.
-func (h *RemoteHost) Init(ctx e2e.Context) error {
+func (h *RemoteHost) Init(ctx e2e.Context) (err error) {
 	h.context = ctx
-	return h.ReconnectSSH()
-}
-
-// Execute executes a command and returns an error if any.
```

Comment:
Note for reviewers:

This part moved to https://github.com/DataDog/datadog-agent/pull/25346/files#diff-59372689c0cc61c5b3cf4e4da1cbf8b84e123467e5628019cde423e3ce68f1b3R104